src/problem/solve_problem.jl

```diff
@@ -115,9 +120,13 @@ function initial_solution(
     end
 
     solution_init::MSTSolution = MSTSolution(cluster_sets, ms_soln_sets, tender_soln_sets)
-    solution_opt::MSTSolution = optimize_waypoints(solution_init, problem)
 
-    return solution_opt
+    if waypoint_optim_flag
+        @info "Optimizing waypoints using gradient descent"
+        return optimize_waypoints(solution_init, problem)
+    else
+        return solution_init
+    end
```

Comment:
There's a pattern known as early return which can be applied here.

Reducing the amount of indentations leads to clearer and more straightforward code.

To wit, the `else` here is unnecessary.

```suggestion
    if waypoint_optim_flag
        @info "Optimizing waypoints using gradient descent"
        return optimize_waypoints(solution_init, problem)
    end
    
    return solution_init
```